gemini/function-calling/parallel_function_calling.ipynb

```diff
@@ -1,814 +1,842 @@
+        "    FunctionDeclaration,\n",
+        "    GenerationConfig,\n",
+        "    GenerationResponse,\n",
+        "    GenerativeModel,\n",
+        "    Part,\n",
+        "    Tool,\n",
+        ")\n",
+        "import wikipedia"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "b2acd610d52c"
+      },
+      "source": [
+        "### Define helper function"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "metadata": {
+        "id": "32c90d8c452a"
+      },
+      "outputs": [],
+      "source": [
+        "# Helper function to extract one or more function calls from a Gemini Function Call response\n",
+        "\n",
+        "\n",
+        "def extract_function_calls(response: GenerationResponse) -> List[Dict]:\n",
+        "    function_calls: List[Dict] = []\n",
+        "    if response.candidates[0].function_calls:\n",
+        "        for function_call in response.candidates[0].function_calls:\n",
+        "            function_call_dict: Dict[str, Dict[str, Any]] = {function_call.name: {}}\n",
+        "            for key, value in function_call.args.items():\n",
+        "                function_call_dict[function_call.name][key] = value\n",
+        "            function_calls.append(function_call_dict)\n",
+        "    return function_calls"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "j3KHAr6BsJY6"
+      },
+      "source": [
+        "## Example: Parallel function calls on the same function\n",
+        "\n",
+        "A great use case for parallel function calling is when you have a function that only accepts one parameter per API call and you need to make repeated calls to that function.\n",
+        "\n",
+        "With Parallel Function Calling, rather than having to send N number of API requests to Gemini for N number function calls, instead you can send a single API request to Gemini, receive N number of Function Call Responses within a single response, make N number of external API calls in your code, then return all of the API responses to Gemini in bulk. And you can do all of this without any extra configuration in your function declarations, tools, or requests to Gemini.\n",
+        "\n",
+        "In this example, you'll do exactly that and use Parallel Function Calling in Gemini to ask about multiple topics on [Wikipedia](https://www.wikipedia.org/). Let's get started!\n",
+        "\n",
+        "### Write function declarations and wrap them in a tool\n",
+        "\n",
+        "First, define your function declarations and tool using the Vertex AI Python SDK:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 7,
+      "metadata": {
+        "id": "31a4ae78030e"
+      },
+      "outputs": [],
+      "source": [
+        "search_wikipedia = FunctionDeclaration(\n",
+        "    name=\"search_wikipedia\",\n",
```

Comment:
Not necessarily in scope for this PR, but would it be good to encourage the practice of setting this name to a `CONSTANT` so that spelling errors aren't as big of an issue when checking later?

Reply:
Thanks for reviewing! Makes sense - I'll consider that in a future update/refresh of these sample notebooks.